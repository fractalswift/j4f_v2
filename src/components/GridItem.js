import styled from "styled-components"
import { Link } from "gatsby"

const GridItem = styled(Link)`
  position: relative;
  > div {
    position: absolute !important;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
  }
  > div img {
    transition: all 0.3s ease 0s !important;
  }
  > span {
    z-index: 10;
    color: white;
    position: absolute;
    left: 0;
    right: 0;
    text-align: right;
    font-weight: 700;
    font-size: 1.875rem;
    padding: 1.5rem;
    text-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
  }

  > p {
    z-index: 10;
    color: white;
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
    text-align: right;
    font-weight: 300;
    font-size: 1.074rem;
    padding: 1.5rem;
    text-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
  }

  &:hover {
    > div img {
      transform: scale(1.1);
    }
  }

  @media (max-width: 1024px) {
    width: 100vw;
    span {
      font-size: 5vh;
    }
  }
`

export default GridItem
