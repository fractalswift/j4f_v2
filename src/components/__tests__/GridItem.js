import React from "react"
import renderer from "react-test-renderer"

import GridItem from "../GridItem"

describe("GridItem", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<GridItem />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
