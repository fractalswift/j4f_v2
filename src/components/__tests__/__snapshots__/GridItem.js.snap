// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem renders correctly 1`] = `
<a
  className="sc-AxjAm JTeds"
/>
`;
